crates/rust-analyzer/src/cli/scip.rs

```diff
@@ -142,7 +142,7 @@ impl flags::Scip {
                 let mut symbol_roles = Default::default();
 
                 if let Some(def) = token.definition {
-                    if def.range == text_range {
+                    if def.file_id == file_id && def.range == text_range {
```

Comment:
do you mind adding a comment above this explaining why this is needed, linking to this PR?